src/operations/execute_operation.ts

```diff
@@ -224,23 +226,25 @@ function executeWithServerSelection<TResult>(
     }
 
     // select a new server, and attempt to retry the operation
-    topology.selectServer(selector, serverSelectionOptions, (error?: Error, server?: Server) => {
-      if (!error && isWriteOperation && !supportsRetryableWrites(server)) {
-        return callback(
-          new MongoUnexpectedServerResponseError(
-            'Selected server does not support retryable writes'
-          )
-        );
-      }
+    setTimeout(() => {
```

Comment:
need to double check if we're actually selecting a server with a paused pool, which should not be possible